### PR TITLE
Requireable WorkerId

### PR DESCRIPTION
**Pull request by paulbalaji**
#### Description

- `WorkerId` struct
- `WorkerIdSubscriptionManager`
- Added a test script to playground to print out the Worker ID

#### Tests

- ran in playground

#### Documentation

- [x] changelog

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.core/Subscriptions/StandardSubscriptionManagers/WorkerIdSubscriptionManager.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        public override void Cancel(ISubscription subscription)
+        {
+        }
+
+        public override void ResetValue(ISubscription subscription)
+        {
+        }
+    }
+
+    public struct WorkerId
```
Comment (zeroZshadow) [inline workers/unity/Packages/com.improbable.gdk.core/Subscriptions/StandardSubscriptionManagers/WorkerIdSubscriptionManager.cs]: Can this be a readonly struct with a readonly field for the Id? The Id is now a property which makes it a function call. Not ideal.

File: workers/unity/Packages/com.improbable.gdk.core/Subscriptions/StandardSubscriptionManagers/WorkerIdSubscriptionManager.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        public WorkerIdSubscriptionManager(World world)
+        {
+            this.world = world;
+        }
+
+        public override Subscription<WorkerId> Subscribe(EntityId entityId)
+        {
+            var subscription = new Subscription<WorkerId>(this, new EntityId(0));
+            subscription.SetAvailable(new WorkerId(world.GetExistingSystem<WorkerSystem>().WorkerId));
```
Comment (zeroZshadow) [inline workers/unity/Packages/com.improbable.gdk.core/Subscriptions/StandardSubscriptionManagers/WorkerIdSubscriptionManager.cs]: Can this be split into 2 lines?

File: workers/unity/Packages/com.improbable.gdk.core/Subscriptions/StandardSubscriptionManagers/WorkerIdSubscriptionManager.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        private readonly World world;
+
+        public WorkerIdSubscriptionManager(World world)
+        {
+            this.world = world;
+        }
+
+        public override Subscription<WorkerId> Subscribe(EntityId entityId)
+        {
+            var subscription = new Subscription<WorkerId>(this, new EntityId(0));
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.core/Subscriptions/StandardSubscriptionManagers/WorkerIdSubscriptionManager.cs]: You can cache this subscription so we don't reallocate this every time we subscribe.

File: workers/unity/Packages/com.improbable.gdk.core/Subscriptions/StandardSubscriptionManagers/WorkerIdSubscriptionManager.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        public override void Cancel(ISubscription subscription)
+        {
+        }
+
+        public override void ResetValue(ISubscription subscription)
+        {
+        }
+    }
+
+    public readonly struct WorkerId
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.core/Subscriptions/StandardSubscriptionManagers/WorkerIdSubscriptionManager.cs]: I'm kind of in favor of providing `implicit` casts to/from string for this. Given that this is basically a type-alias.

File: workers/unity/Assets/Playground/Scripts/MonoBehaviours/WorkerIdTest.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+using Improbable.Gdk.Subscriptions;
+using UnityEngine;
+
+namespace Playground
+{
+    public class WorkerIdTest : MonoBehaviour
```
Comment (jessicafalk) [inline workers/unity/Assets/Playground/Scripts/MonoBehaviours/WorkerIdTest.cs]: could we make a proper test out of this?

**Pull request merged**